src/vs/workbench/contrib/timeline/common/timelineService.ts

```diff
@@ -60,6 +60,8 @@ export class TimelineService implements ITimelineService {
 			return undefined;
 		}
 
+		const disposables = new DisposableStore();
+		disposables.add(tokenSource);
```

Comment:
This seems wrong: the token source was not created in here, yet is being tracked for disposal?